SegmentationModule.py

```diff
@@ -0,0 +1,238 @@
+import numpy as np
+from keras.callbacks import ModelCheckpoint, LearningRateScheduler, EarlyStopping
+from astropy.convolution import Gaussian1DKernel, convolve
+import datetime
+
+import SegmSubmodules.Models as Models
+import SegmSubmodules.Evalutation as Eval
+import SegmSubmodules.PredictionCutter as Cutter
+
+
+class SegmentationModule:
+    """This class represents pipeline module responsible for segmentation"""
+
+    # public:
+    def __init__(self):
+        """Initializes module"""
+        self.model = None
+
+    def exec_fit(self, x_train, x_valid, y_train, y_valid, checkpoint_file, epochs=30, batch_size=32):
+        """Creates model and launches fit. Trained model stored in self.model
+
+        Args:
+            x_train: train data tensor (number of samples, duration, embedding)
+            x_valid: validation data tensor
+            y_train: train data labels tensor
+            y_valid: validation data labels tensor
+            checkpoint_file: name of file where checkpoints will be saved
+            epochs: the number of epochs to fit
+            batch_size: the number of samples in a single batch to fit"""
+        assert len(x_train.shape) == 3, "X_train shape must be (samples, time, embeddings)"
+        assert len(x_valid.shape) == 3, "X_valid shape must be (samples, time, embeddings)"
+        assert len(y_train.shape) == 3, "Y_train shape must be (samples, time, 1)"
+        assert len(y_valid.shape) == 3, "Y_valid shape must be (samples, time, 1)"
+        assert isinstance(checkpoint_file, str) and checkpoint_file.endswith(".h5"), ("Checkpoint file must be string"
+                                                                                      "ends with .h5")
+        if self.model is None:  # build default model
+            self.__build_new_model()
+        y_tr_crf, y_val_crf = Models.data_to_crf(y_train, y_valid)  # convert to CRF-suitable format
```

Comment:
name method `convert_to_crf_format` or like that and there will be no need of comment)